benches/benches.rs

```diff
@@ -14,24 +10,70 @@ use cosmian_findex::InMemory;
 
 #[cfg(feature = "sqlite-mem")]
 use cosmian_findex::SqliteMemory;
+#[cfg(feature = "sqlite-mem")]
+const SQLITE_PATH: &str = "./target/benches.sqlite";
 
 #[cfg(feature = "redis-mem")]
 use cosmian_findex::RedisMemory;
+/// Gets the Redis URL to use based on the following priority:
+/// 1. If REDIS_HOST environment variable is set, use "redis://{REDIS_HOST}:6379"
+/// 2. If no environment variable is set but a Redis server is reachable at
+///    localhost (127.0.0.1:6379), use "redis://127.0.0.1:6379"
+/// 3. If neither of the above conditions are met, fall back to the default
+///    dockered Redis service at "redis://redis:6379"
+#[cfg(feature = "redis-mem")]
+fn get_redis_url() -> String {
+    if let Ok(host) = std::env::var("REDIS_HOST") {
+        return format!("redis://{}:6379", host);
+    }
 
-// Number of points in each graph.
-const N_PTS: usize = 9;
+    use std::{net::TcpStream, time::Duration};
+    match TcpStream::connect_timeout(
+        &"127.0.0.1:6379".parse().unwrap(),
+        Duration::from_millis(20),
+    ) {
+        Ok(_) => "redis://127.0.0.1:6379".to_string(),
+        Err(_) => "redis://redis:6379".to_string(),
+    }
+}
 
-#[cfg(feature = "sqlite-mem")]
-const SQLITE_PATH: &str = "./target/benches.sqlite";
+/// To run the postgresql benchmarks locally, add the following service to your pg_service.conf file (usually under ~/.pg_service.conf):
+/// [cosmian_service]
+/// host=localhost
+/// dbname=cosmian
+/// user=cosmian
+/// password=cosmian
```

Comment:
This doc should be next to the PgSQL memory implementation. Since anyway it is need to pass the tests.